Modules/Forum/classes/class.ilFileDataForum.php

```diff
@@ -11,73 +11,59 @@
 */
 class ilFileDataForum extends ilFileData
 {
-    /**
-    * obj_id
-    * @var integer obj_id of exercise object
-    * @access private
-    */
-    public $obj_id;
-    public $pos_id;
+    const FORUM_PATH = 'forum';
+    public int $obj_id;
+    public int $pos_id;
 
-    /**
-    * path of exercise directory
-    * @var string path
-    * @access private
-    */
-    public $forum_path;
+    public string $forum_path;
     
-    private $error;
+    private mixed $error;
```

Comment:
We MUST NOT use the explicit `mixed` type. The code MUST be compatible with PHP 7.4.